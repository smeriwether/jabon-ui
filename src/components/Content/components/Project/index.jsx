import React, { Component } from 'react';
import PropTypes from 'prop-types';

import ExpandDown from '../../../../UI/ExpandDown';

const propTypes = {
  project: PropTypes.shape({
    wsdlLink: PropTypes.string,
    header1: PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string
    }),
    header2: PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string
    })
  }),
  onChange: PropTypes.func.isRequired,
  onExpand: PropTypes.func.isRequired
};

const defaultProps = {
  project: {
    wsdlLink: '',
    header1: { key: '', value: '' },
    header2: { key: '', value: '' }
  }
};

const styles = {
  label: 'fw7 f6',
  input: 'w-100 pa2 mt2 br2 b--black-20 ba f6'
};

const text = {
  placeholder: 'Placeholder...'
};

class Project extends Component {
  constructor(props) {
    super(props);

    this.state = {
      wsdlLink: props.project.wsdlLink,
      header1Key: props.project.header1.key,
      header1Value: props.project.header1.value,
      header2Key: props.project.header2.key,
      header2Value: props.project.header2.value
    };
  }

  onWsdlChange = (input) => {
    const value = input.target.value;
    this.setState({ wsdlLink: value });
    this.props.onChange({ wsdlLink: value });
  }

  onHeader1KeyChange = (input) => {
    const value = input.target.value;
    this.setState({ header1Key: value });
    this.props.onChange({ header1: { key: value, value: this.state.header1Value } });
  }

  onHeader1ValueChange = (input) => {
    const value = input.target.value;
    this.setState({ header1Value: value });
    this.props.onChange({ header1: { key: this.state.header1Key, value: value } });
  }

  onHeader2KeyChange = (input) => {
    const value = input.target.value;
    this.setState({ header2Key: value });
    this.props.onChange({ header2: { key: value, value: this.state.header2Value } });
  }

  onHeader2ValueChange = (input) => {
    const value = input.target.value;
    this.setState({ header2Value: value });
    this.props.onChange({ header2: { key: this.state.header2Key, value: value } });
  }

  render() {
    return (
      <div className='h-100 w-100 pa3 near-black overflow-scroll'>
        <div className='flex justify-between'>
          <div className='b f3'>
            Project Information
          </div>
          <div className='pointer' data-role='expand' onClick={ this.props.onExpand } >
            <ExpandDown />
          </div>
        </div>

        <div className='pt4 pb4'>
          <label className={ styles.label } htmlFor='wsdl'>WSDL Link</label>
          <input
            id='wsdl'
            data-role='wsdl'
            value={ this.state.wsdlLink }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onWsdlChange(i) } }
          >
          </input>
        </div>

        <div className='w-50 fl pr3 pb3'>
          <label className={ styles.label } htmlFor='header1-key'>Header Key</label>
          <input
            id='header1-key'
            data-role='header1-key'
            value={ this.state.header1Key }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onHeader1KeyChange(i) } }
          >
          </input>
        </div>
        <div className='w-50 fl pb3'>
          <label className={ styles.label } htmlFor='header1-value'>Header Value</label>
          <input
            id='header1-value'
            data-role='header1-value'
            value={ this.state.header1Value }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onHeader1ValueChange(i) } }
          >
          </input>
        </div>

        <div className='w-50 fl pr3 pb3'>
          <label className={ styles.label } htmlFor='header2-key'>Header Key</label>
          <input
            id='header2-key'
            data-role='header2-key'
            value={ this.state.header2Key }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onHeader2KeyChange(i) } }
          >
          </input>
        </div>
        <div className='w-50 fl pb3'>
          <label className={ styles.label } htmlFor='header2-value'>Header Value</label>
          <input
            id='header2-value'
            data-role='header2-value'
            value={ this.state.header2Value }
            className={ styles.input }
            placeholder={ text.placeholder }
            onChange={ (i) => { this.onHeader2ValueChange(i) } }
          >
          </input>
        </div>
      </div>
    );
  }
};

Project.propTypes = propTypes;
Project.defaultProps = defaultProps;

export default Project;
