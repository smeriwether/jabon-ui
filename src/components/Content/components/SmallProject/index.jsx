import React from 'react';
import PropTypes from 'prop-types';

import ExpandUp from '../../../../UI/ExpandUp';

const propTypes = {
  onExpand: PropTypes.func.isRequired
};

const SmallProject = ({ onExpand }) => {
  return (
    <div className='h-100 w-100 near-black flex items-center justify-between'>
      <div className='b f3 pa2'>
        Project Information
      </div>
      <div className='pointer pa2' data-role='expand' onClick={ onExpand } >
        <ExpandUp />
      </div>
    </div>
  );
};

SmallProject.propTypes = propTypes;

export default SmallProject;
