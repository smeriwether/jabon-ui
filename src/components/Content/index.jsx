import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Editor from '../../components/Editor';
import Project from './components/Project';
import SmallProject from './components/SmallProject';

const propTypes = {
  content: PropTypes.string,
  project: PropTypes.shape({
    wsdlLink: PropTypes.string,
    header1: PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string
    }),
    header2: PropTypes.shape({
      key: PropTypes.string,
      value: PropTypes.string
    })
  }),
  onContentChange: PropTypes.func.isRequired,
  onProjectChange: PropTypes.func.isRequired,
  expanded: PropTypes.bool
};

const defaultProps = {
  expanded: true,
  project: {
    wsdlLink: '',
    header1: { key: '', value: '' },
    header2: { key: '', value: '' }
  }
};

const classes = {
  expandedEditor: 'h-65 w-100 fl',
  shrinkedEditor: 'h-100-h50 w-100 fl',
  expandedProject: 'h-35 w-100 fl bb bt b--moon-gray',
  shrinkedProject: 'h50 w-100 fl bb bt b--moon-gray'
};

class Content extends Component {
  constructor(props) {
    super(props);

    this.state = {
      project: {
        wsdlLink: props.project.wsdlLink,
        header1: props.project.header1,
        header2: props.project.header2
      },
      expanded: props.expanded
    };
  }

  toggleExpand = () => {
    this.setState((prevState) => ({
      ...prevState,
      expanded: !prevState.expanded
    }));
  }

  onProjectChange = (project) => {
    this.setState((prevState) => ({
      ...prevState,
      project: Object.assign(prevState.project, project)
    }));
    this.props.onProjectChange(project);
  }

  render() {
    const { content, onContentChange } = this.props;

    return (
      <div className='h-100 w-100 br b--moon-gray'>
        <div className={ this.state.expanded ? classes.expandedEditor : classes.shrinkedEditor }>
          <Editor value={ content } name='main-editor' onChange={ onContentChange } />
        </div>
        {
          this.state.expanded ?
            <div className={ classes.expandedProject }>
              <Project
                project={ this.state.project }
                onExpand={ this.toggleExpand }
                onChange={ this.onProjectChange }
              />
            </div>
          :
            <div className={ classes.shrinkedProject }>
              <SmallProject onExpand={ this.toggleExpand } />
            </div>
        }
      </div>
    );
  }
}

Content.propTypes = propTypes;
Content.defaultProps = defaultProps;

export default Content;
