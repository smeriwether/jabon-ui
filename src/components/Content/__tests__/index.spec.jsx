import Content from '../../Content';

import Project from '../components/Project';
import SmallProject from '../components/SmallProject';

describe('<Content />', () => {
  const setup = (overrides) => {
    const props = Object.assign({
      content: null,
      onContentChange: jest.fn(),
      onProjectChange: jest.fn(),
      expanded: true
    }, overrides);

    const wrapper = shallow(<Content { ...props } />);

    return { wrapper };
  };

  context('when expanded', () => {
    it('should match the snapshot', () => {
      const { wrapper } = setup({ content: 'Some content', expanded: true });
      expect(wrapper).toMatchSnapshot();
    });

    it('should show the Project component', () => {
      const { wrapper } = setup({ content: 'Some content', expanded: true });
      expect(wrapper.find(Project).length).toEqual(1);
    });
  });

  context('when not expanded', () => {
    it('should match the snapshot', () => {
      const { wrapper } = setup({ content: 'Some content', expanded: false });
      expect(wrapper).toMatchSnapshot();
    });

    it('should show the SmallProject component', () => {
      const { wrapper } = setup({ content: 'Some content', expanded: false });
      expect(wrapper.find(SmallProject).length).toEqual(1);
    });
  });

  describe('onProjectChange', () => {
    it('should call the onProjectChange prop function', () => {
      const onProjectChangeMock = jest.fn();
      const { wrapper } = setup({ onProjectChange: onProjectChangeMock });

      const project = { foo: 'bar' };
      wrapper.instance().onProjectChange(project);

      expect(onProjectChangeMock).toHaveBeenCalled();
      expect(onProjectChangeMock).toHaveBeenCalledWith(project);
    });
  });
});
