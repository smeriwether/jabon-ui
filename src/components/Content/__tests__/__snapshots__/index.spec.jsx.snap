// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> when expanded should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    content="Some content"
    expanded={true}
    onContentChange={[MockFunction]}
    onProjectChange={[MockFunction]}
    project={
      Object {
        "header1": Object {
          "key": "",
          "value": "",
        },
        "header2": Object {
          "key": "",
          "value": "",
        },
        "wsdlLink": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="h-65 w-100 fl"
        >
          <Editor
            name="main-editor"
            onChange={[MockFunction]}
            readOnly={false}
            value="Some content"
          />
        </div>,
        <div
          className="h-35 w-100 fl bb bt b--moon-gray"
        >
          <Project
            onChange={[Function]}
            onExpand={[Function]}
            project={
              Object {
                "header1": Object {
                  "key": "",
                  "value": "",
                },
                "header2": Object {
                  "key": "",
                  "value": "",
                },
                "wsdlLink": "",
              }
            }
          />
        </div>,
      ],
      "className": "h-100 w-100 br b--moon-gray",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Editor
            name="main-editor"
            onChange={[MockFunction]}
            readOnly={false}
            value="Some content"
          />,
          "className": "h-65 w-100 fl",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "main-editor",
            "onChange": [MockFunction],
            "readOnly": false,
            "value": "Some content",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Project
            onChange={[Function]}
            onExpand={[Function]}
            project={
              Object {
                "header1": Object {
                  "key": "",
                  "value": "",
                },
                "header2": Object {
                  "key": "",
                  "value": "",
                },
                "wsdlLink": "",
              }
            }
          />,
          "className": "h-35 w-100 fl bb bt b--moon-gray",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "onExpand": [Function],
            "project": Object {
              "header1": Object {
                "key": "",
                "value": "",
              },
              "header2": Object {
                "key": "",
                "value": "",
              },
              "wsdlLink": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="h-65 w-100 fl"
          >
            <Editor
              name="main-editor"
              onChange={[MockFunction]}
              readOnly={false}
              value="Some content"
            />
          </div>,
          <div
            className="h-35 w-100 fl bb bt b--moon-gray"
          >
            <Project
              onChange={[Function]}
              onExpand={[Function]}
              project={
                Object {
                  "header1": Object {
                    "key": "",
                    "value": "",
                  },
                  "header2": Object {
                    "key": "",
                    "value": "",
                  },
                  "wsdlLink": "",
                }
              }
            />
          </div>,
        ],
        "className": "h-100 w-100 br b--moon-gray",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Editor
              name="main-editor"
              onChange={[MockFunction]}
              readOnly={false}
              value="Some content"
            />,
            "className": "h-65 w-100 fl",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "main-editor",
              "onChange": [MockFunction],
              "readOnly": false,
              "value": "Some content",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Project
              onChange={[Function]}
              onExpand={[Function]}
              project={
                Object {
                  "header1": Object {
                    "key": "",
                    "value": "",
                  },
                  "header2": Object {
                    "key": "",
                    "value": "",
                  },
                  "wsdlLink": "",
                }
              }
            />,
            "className": "h-35 w-100 fl bb bt b--moon-gray",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "onExpand": [Function],
              "project": Object {
                "header1": Object {
                  "key": "",
                  "value": "",
                },
                "header2": Object {
                  "key": "",
                  "value": "",
                },
                "wsdlLink": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Content /> when not expanded should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    content="Some content"
    expanded={false}
    onContentChange={[MockFunction]}
    onProjectChange={[MockFunction]}
    project={
      Object {
        "header1": Object {
          "key": "",
          "value": "",
        },
        "header2": Object {
          "key": "",
          "value": "",
        },
        "wsdlLink": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="h-100-h50 w-100 fl"
        >
          <Editor
            name="main-editor"
            onChange={[MockFunction]}
            readOnly={false}
            value="Some content"
          />
        </div>,
        <div
          className="h50 w-100 fl bb bt b--moon-gray"
        >
          <SmallProject
            onExpand={[Function]}
          />
        </div>,
      ],
      "className": "h-100 w-100 br b--moon-gray",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Editor
            name="main-editor"
            onChange={[MockFunction]}
            readOnly={false}
            value="Some content"
          />,
          "className": "h-100-h50 w-100 fl",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "main-editor",
            "onChange": [MockFunction],
            "readOnly": false,
            "value": "Some content",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SmallProject
            onExpand={[Function]}
          />,
          "className": "h50 w-100 fl bb bt b--moon-gray",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onExpand": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="h-100-h50 w-100 fl"
          >
            <Editor
              name="main-editor"
              onChange={[MockFunction]}
              readOnly={false}
              value="Some content"
            />
          </div>,
          <div
            className="h50 w-100 fl bb bt b--moon-gray"
          >
            <SmallProject
              onExpand={[Function]}
            />
          </div>,
        ],
        "className": "h-100 w-100 br b--moon-gray",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Editor
              name="main-editor"
              onChange={[MockFunction]}
              readOnly={false}
              value="Some content"
            />,
            "className": "h-100-h50 w-100 fl",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "main-editor",
              "onChange": [MockFunction],
              "readOnly": false,
              "value": "Some content",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SmallProject
              onExpand={[Function]}
            />,
            "className": "h50 w-100 fl bb bt b--moon-gray",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onExpand": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
